pkg/subscriptions/subscriptions.js

```diff
@@ -0,0 +1,502 @@
+             * subscriptions might also be invalid (e.g. expired), but we don't handle that currently
+             */
+            if (overall_status !== 'Unknown') {
+                system_unregistered.hide();
+            }
+            else {
+                /* no valid subscription: hide output and present user with the option to register system */
+                subscription_info.hide();
+                system_unregistered.show();
+            }
+
+            /* if refresh was requested, try again - otherwise get details */
+            if (status_update_requested)
+                update_subscriptions();
+            else if (current_system_status !== 'Unknown')
+                get_subscription_details();
+        }
+
+        /* get subscription summary using 'subscription-manager status'*/
+        function get_subscription_status() {
+            if (action_in_progress()) {
+                status_update_requested = true;
+                return;
+            }
+            status_update_requested = false;
+
+            show_progress_message('Retrieving current system subscription status.');
+
+            /* we need a buffer for 'subscription-manager status' output, since that can fail with a return code != 0
+             * even if we need its output (no valid subscription)
+             */
+            var status_buffer = '';
+            /* TODO DBus API doesn't deliver what we need, so we call subscription manager
+             * without translations and parse the output
+             *
+             * 'subscription-manager status' will only return with exit code 0 if all is well (and subscriptions current)
+             */
+            cockpit.spawn(['subscription-manager', 'status'],
+                    { directory: '/', superuser: true, environ: ['LC_ALL=C'] })
+                .stream(function(text) {
+                    status_buffer += text;
+                    return text.length;
+                }).done(function(text) {
+                    process_status_output(status_buffer + text, '');
+                }).fail(function(ex) {
+                    process_status_output(status_buffer, ex);
+                });
+        }
+
+        /* request update via DBus */
+        function update_subscriptions() {
+            status_update_requested = false;
+            var call = service.call(
+                '/EntitlementStatus',
+                'com.redhat.SubscriptionManager.EntitlementStatus',
+                'check_status',
+                []);
+
+            call.fail(function(ex) {
+                    console.warn("Subscriptions: couldn't request update: " + ex);
+                });
+        }
+
+        function register_system(url, username, password) {
+            if (action_in_progress()) {
+                $('#subscriptions-register-failure-reason').text(
+                  'Unable to register at this time because a call to subscription manager ' +
```

Comment:
`subscriptions-register-failure` doesn't exist anymore, so I changed this into a console log message. This shouldn't happen on a ui-path anyway: the "Register" button is hidden during status check and while performing registration.